examples/kyb-app/src/components/organisms/UIRenderer/elements/StepperUI/StepperUI.tsx

```diff
@@ -6,15 +6,41 @@ import { usePageResolverContext } from '@app/components/organisms/DynamicUI/Page
 import { useStateManagerContext } from '@app/components/organisms/DynamicUI/StateManager/components/StateProvider';
 import { useDynamicUIContext } from '@app/components/organisms/DynamicUI/hooks/useDynamicUIContext';
 import { useMemo } from 'react';
+import { usePageContext } from '@app/components/organisms/DynamicUI/Page';
+import { UIPage } from '@app/domains/collection-flow';
+import { UIElementState } from '@app/components/organisms/DynamicUI/hooks/useUIStateLogic/hooks/useUIElementsStateLogic/types';
+import { ErrorField } from '@app/components/organisms/DynamicUI/rule-engines';
 
 export const StepperUI = () => {
   const { state: uiState } = useDynamicUIContext();
   const { pages, currentPage } = usePageResolverContext();
   const { state } = useStateManagerContext();
+  const { pageErrors } = usePageContext();
+  const computeStepStatus = ({
+    uiElement,
+    page,
+    pageError,
+    currentPage,
+  }: {
+    uiElement: UIElementState;
+    page: UIPage;
+    pageError: Record<string, ErrorField>;
+    currentPage: UIPage;
+  }) => {
+    if (!!Object.keys(pageError).length && currentPage.number === page.number) return 'warning';
+    if (uiElement?.isCompleted) return 'completed';
```

Comment:
can you describe this one? I am not really getting it

Reply:
Only line 30 is new. `pageError` may be an empty object here, so I'm both checking if `pageError` has keys and if the breadcrumb is for the current step.

Reply:
gotcha, cool